### DISCUSSION TITLE
Allow multiple tax rates for each product

### DISCUSSION THREAD

**Discussion by jonassiewertsen** [jonassiewertsen/statamic-butik#164] [Show and tell]
As pointet out by @nn-dmt, multiple tax rates per product would be really helpful: https://github.com/jonassiewertsen/statamic-butik/issues/162

> You can only select one tax rate on product.... in order for this to be useful to Americans, the tax rate needs to be tied to the shipping location and update accordingly based on each individual US State, who each have their own tax rates/laws.
> 
> I can only setup a general 'Maximum possible' tax on the products, but that would set the tax of each product over 8.75% when some states have ZERO tax, and most states have 7.

**Is this a problem for more people?**

How should it work to be most flexible?

**Comment by ghost** [comment]
Hi,

Sorry for the confusion listing it as an issue.

This is going to be a problem for every single US vendor/developer - and as it stands I cannot use the product correctly. I am presently going to charge 0% tax and build a maximum possible tax into the cost of the item, which is not ideal as customers need transparency on the money they are spending.

It is because of how every state in the country handles tax rates differently - some states charge no tax, some charge a base amount but then an additional premium percentage based on type of item (if it food it is taxed differently than toys, where services generally are not taxed) - 

Ideally, the tax rate would need to be tied to the location variable and the user can keep a spreadsheet or list or collection with tax rates for each state. I created tax rates for each state in butik, but there is no way to dynamically insert the tax onto the item at the time of adding shipping - so I can only set a default single tax rate on a single item, or have a different version of the item for each state/tax rate... you can see how that would be tedious!

The least tedious solution for you is to tie the tax rate to dynamically update only after the billing/shipping address has populated, and the user can keep the maintained list of tax rates as their responsibility since the states will change laws and rates over time...

To give you an example of taxes here, California has a base tax of 7.25%, but some counties in California do an additional 1.5% based on local fees - in my home town, tax is 8.75% on tangible goods. So unfortunately the only workable solution, short of keeping a dynamically populated list of taxes (companies make whole business models out of this with woo commerce), is to do a 'maximum tax' value for each state. I am fine doing that, it is how many companies do e-commerce online here. However the way the system works, even that is not possible.

This is going to prevent all US vendors from using the product unless there is some form of solution to workaround it, otherwise we will all need to do a max tax of over 12%, or no tax. Both are not optimal solutions.

**Comment by ghost** [comment]
Here is more information on the economic tax nexus laws here in the USA (which were introduced fairly recently) -

```https://www.avalara.com/us/en/learn/nexus.html```

Any company doing large volumes of ecommerce, or certain types of inter state commerce, are required to register in that state and remit tax to that state... it's a huge mess. The company linked above, alavara, has a whole business model based on servicing that tax-rate monitoring need within software.

**Comment by freshface** [comment]
Hi

I am from belgium and if we sell to outside our own country no vat or tax should be applied.

**Reply by freshface** [reply]
Update on this comment.
The tax changes in the cart view if the shipping cost changes (if you change the country).
The shipping cost has no tax rate applied and should not be changed.

This is checked with our accountant.

**Reply by jonassiewertsen** [reply]
If the shipping cost has not tax rate applied or if they do, is different from country to country, so the implementation needs to cover multiple options.

**Comment by jonassiewertsen** [comment]
Multi taxes have been added to butik version 4, which is still Work in Progress. 

Multi taxes will be country base at first. If state based multi taxes are a thing, please open a new discussion for it. 

![Screenshot_2021-03-31_at_23 18 39](https://user-images.githubusercontent.com/38906163/113625925-d2596480-9661-11eb-8411-e4caacb1ab6e.png)
